### PR TITLE
Jb39150

### PR DESCRIPTION
**Pull request by jlehtoranta**
Add support for build engine proxy configuration

### CODE REVIEW

File: src/plugins/mer/mersdkdetailswidget.cpp [hunk 1]
```diff
@@ -208,5 +234,142 @@ void MerSdkDetailsWidget::onPathChooserEditingFinished()
...
+
+    if (servers == m_wwwProxyServers) {
+        return;
+    }
+
+    if (m_wwwProxy == MER_SDK_PROXY_AUTOMATIC) {
+        reg.setPattern("^(\\S*)(.*)$");
+        servers.replace(reg, "\\1");
+        reg.setPattern("[,;]");
+        servers.replace(reg, "");
```
Comment (martyone) [inline src/plugins/mer/mersdkdetailswidget.cpp]: I do understand the regexps, but I do not get the purpose :) Could you please add a comment?

**Comment by jlehtoranta** [comment]
PR cleaned up + decided to keep the hide/show behavior of the proxy options.

File: src/plugins/mer/mersdkdetailswidget.cpp [hunk 2]
```diff
@@ -208,5 +233,131 @@ void MerSdkDetailsWidget::onPathChooserEditingFinished()
...
+    QString proxies = s;
+    QRegularExpression reg;
+
+    int cpos = m_ui->wwwProxyServersLine->cursorPosition();
+    int l = proxies.length();
+
+    reg.setPattern("[,;\\s]");
+    if (m_wwwProxy == MER_SDK_PROXY_AUTOMATIC) {
+        // Allow to input a single item with no spaces, commas or semicolons
+        proxies.replace(reg, "");
```
Comment (martyone) [inline src/plugins/mer/mersdkdetailswidget.cpp]: Wouldn't this break if the URL contains comma or semicolon? Sometimes these are used in tuple-like parameter values. Maybe cleaning whitespaces would be sufficient.

Comment (jlehtoranta) [reply src/plugins/mer/mersdkdetailswidget.cpp]: Fixed by strong quoting in wwwproxycommand.cpp. All characters are now allowed for proxy autoconfiguration URL. In manual mode entries are separated by single spaces, so all characters except whitespaces are allowed for each entry.

**Pull request merged**